Avoid emitting a reference to an undefined symbol.

Without this, linking might fail with an error like the following with
gfortran 16.2.0:

FAILED: [code=1] Src/libsiesta.dll Src/libsiesta.dll.a
C:\WINDOWS\system32\cmd.exe /C "cd . && C:\msys64\ucrt64\bin\gfortran.exe -fallow-invalid-boz -fallow-argument-mismatch -march=x86-64-v3 -fallow-argument-mismatch -O3  -shared -o Src\libsiesta.dll -Wl,--out-implib,Src\libsiesta.dll.a -Wl,--major-image-version,0,--minor-image-version,0 @CMakeFiles\SIESTA.libsiesta.rsp && cd ."
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/16.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: Src/CMakeFiles/SIESTA.libsiesta.dir/kpoint_t.F90.obj:kpoint_t.F90:(.text+0x3f28): undefined reference to `process_k_cell_'
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/16.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: Src/CMakeFiles/SIESTA.libsiesta.dir/kpoint_t.F90.obj:kpoint_t.F90:(.text+0x4498): undefined reference to `process_k_cell_'
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/16.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: Src/CMakeFiles/SIESTA.libsiesta.dir/kpoint_t.F90.obj:kpoint_t.F90:(.text+0x4705): undefined reference to `process_k_cell_'
collect2.exe: error: ld returned 1 exit status

NOTE: With this patch, no custom subroutines that are passed through the
affected interfaces will be executed.

diff -urN siesta-5.4.1/Src/kpoint_t.F90.orig siesta-5.4.1/Src/kpoint_t.F90
--- siesta-5.4.1/Src/kpoint_t.F90.orig	2025-09-16 14:23:18.000000000 +0200
+++ siesta-5.4.1/Src/kpoint_t.F90	2026-08-30 16:53:29.398120600 +0200
@@ -299,10 +299,6 @@
     subroutine setup_mp()
       use m_find_kgrid, only: find_kgrid
 
-      if ( present(process_k_cell) ) then
-        call process_k_cell(this%k_cell, this%k_displ)
-      end if
-
       ! Find the grid
       call find_kgrid(cell, this%k_cell, this%k_displ, .true., &
           this%trs, &
@@ -343,10 +339,6 @@
 
       end if
 
-      if ( present(process_k_cell) ) then
-        call process_k_cell(this%k_cell, this%k_displ)
-      end if
-
       ! Find the grid
       call find_kgrid(cell,this%k_cell, this%k_displ, .false., &
           this%trs, &
